src/main/java/com/icthh/xm/uaa/lep/XmUaaLepProcessingApplicationListener.java

```diff
@@ -33,35 +37,35 @@
 @RequiredArgsConstructor
 public class XmUaaLepProcessingApplicationListener extends SpringLepProcessingApplicationListener {
 
-    private final AccountService accountService;
-    private final TenantConfigService tenantConfigService;
-    private final UserLoginService userLoginService;
-    private final ProfileEventProducer profileEventProducer;
-
+    private final JwtTokenStore jwtTokeStore;
     private final MailService mailService;
     private final UserService userService;
     private final RestTemplate restTemplate;
-
     private final CommonsService commonsService;
+    private final AccountService accountService;
+    private final UserLoginService userLoginService;
+    private final TenantConfigService tenantConfigService;
+    private final ProfileEventProducer profileEventProducer;
+    private final CustomAuthorizationCodeServices customAuthorizationCodeServices;
+
 
     @Override
     protected void bindExecutionContext(final ScopedContext executionContext) {
 
-        // services
         Map<String, Object> services = new HashMap<>();
 
+        services.put(BINDING_SUB_KEY_SERVICE_USER, userService);
+        services.put(BINDING_SUB_KEY_SERVICE_MAIL, mailService);
+        services.put(BINDING_SUB_KEY_SERVICE_JWT_TOKEN_STORE, jwtTokeStore);
```

Comment:
@anton-contributor 
Great feature!
Just one point worrying me is that you have added JwtTokenService to LEP context. 
This class is from external code base and thus can not be controlled during libraries upgrade. So it may cause problems during further updates for Xm core in relation to LEPs.

i suggest to add wrapping service and put it's interface to Lep context.
Also i've prepared guideline to void misunderstanding ti future: 
https://github.com/xm-online/xm-online/wiki/How-to-create-new-XM-microservice-like-project#lep-context-population-guidelines

Reply:
Hi @amedvedchuk thanks for letting know about this situation!  I added a service wrapper.